src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4232,3 +4285,102 @@ const QList<QgsExpressionFunction *> &QgsExpression::Functions()
   }
   return sFunctions;
 }
+
+QgsWithVariableExpressionFunction::QgsWithVariableExpressionFunction()
+  : QgsExpressionFunction( QStringLiteral( "with_variable" ), 3, QCoreApplication::tr( "General" ) )
+{
+
+}
+
+bool QgsWithVariableExpressionFunction::isStatic( const QgsExpressionNodeFunction *node, QgsExpression *parent, const QgsExpressionContext *context ) const
+{
+  bool isStatic = false;
+
+  QgsExpressionNode::NodeList *args = node->args();
+
+  if ( args->count() < 3 )
+    return false;
+
+  // We only need to check if the node evaluation is static, if both - name and value - are static.
+  if ( args->at( 0 )->isStatic( parent, context ) && args->at( 1 )->isStatic( parent, context ) )
+  {
+    QVariant name = args->at( 0 )->eval( parent, context );
+    QVariant value = args->at( 1 )->eval( parent, context );
+
+    // Temporarily append a new scope to provide the variable
+    appendTemporaryVariable( context, name.toString(), value );
+    if ( args->at( 2 )->isStatic( parent, context ) )
+      isStatic = true;
+    popTemporaryVariable( context );
+  }
+
+  return false;
```

Comment:
I get a build warning here that isStatic is set but never used - and that seems to be true. Should this be returning isStatic instead of always returning false? Otherwise this whole method could be replaced by a single `return false`.